activity-stream/send-to-device-rocket.html

```diff
@@ -0,0 +1,509 @@
+          <span id="error" class="hidden"><span class="alert">!</span> {{ error_text|default('There was an error sending the email. Please try again.', true) }}</span>
+          <p class="disclaimer">
+            <input type="checkbox" id="privacy" required />
+            <label for="privacy">{{ form_disclaimer_html|default('I am okay with Mozilla handling my information as explained in this <a href="https://www.mozilla.org/privacy">Privacy Notice</a>', true)|safe }}</label>
+          </p>
+        </form>
+      </div><!--/.section-content-->
+    </section>
+  </div><!--/#content-wrapper-->
+
+  <button type="button" class="block-snippet-button" title="{{ block_button_text|default('Remove this', true) }}"></button>
+</div>
+
+<script type="text/javascript">
+  //<![CDATA[
+  (function() {
+    const snippetsContainer = document.getElementById('snippets-container');
+    const snippet = document.getElementById('snippet');
+    const style = document.documentElement.style;
+
+    snippet.addEventListener('show_snippet', function () {
+      const input = document.getElementById('email');
+      const showFormButton = document.getElementById('show-form-button');
+      const locale = '{{ locale|default("en-US", true) }}';
+      const intro = document.getElementById('intro');
+      const sendTo = document.getElementById('send-to');
+      const formSubmit = document.getElementById('form-submit');
+      const sendToDeviceForm = document.getElementById('send-to-device-form');
+      const error = document.getElementById('error');
+
+      // compensates for padding in containing element
+      const introHeightModifier = 10;
+
+      // set each time input field is modified/per keystroke
+      let inputOk; 
+
+      // get height of #snippets-container & #intro
+      let scHeight;
+      let introHeight;
+
+      // fire this ASAP after page load
+      setTimeout(() => {
+        // this class only adds transition to transform values so the line below results in a smooth slide up
+        snippetsContainer.classList.add('visible');
+
+        repositionSnippetContainer();
+      }, 500);
+
+      // used for monitoring window resizes
+      // https://davidwalsh.name/javascript-debounce-function
+      function debounce(func, wait, immediate) {
+        var timeout;
+        return function () {
+          var context = this, args = arguments;
+          var later = function () {
+            timeout = null;
+            if (!immediate) func.apply(context, args);
+          };
+          var callNow = immediate && !timeout;
+          clearTimeout(timeout);
+          timeout = setTimeout(later, wait);
+          if (callNow) func.apply(context, args);
+        };
+      };
+
+      // make sure #snippet-container is positioned so the form is hidden just below the bottom of the window
+      function repositionSnippetContainer(specifiedOffset) {
```

Comment:
What you have here totally works so may not be worth changing, but could you avoid all the resize awareness logic if you toggled a hidden div within the snippet container, and then animated it's `max-height` using a CSS keyframe animation? 

Disclaimer: I have no idea how snippet sizing is calculated and have not tested this at all, but if the browser can do the work here it could save some bytes?

Reply:
This is a good approach to try, but I'm going to pass for now due to time constraints.